File: user-service/Application/DTOs/Message/ReadMessageDto.cs
using Domain.Enums;

namespace Application.DTOs.Message;

public class ReadMessageDto
{
    public Guid Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public required string Text { get; set; }
    public required Guid UserId { get; set; }
    public required Guid ChatId { get; set; }
    public required MessageStatus Status { get; set; }
}

File: frontend/src/lib/dto/message/ReadMessageDto.ts
import { MessageStatus } from "../../types/enums";
import { MessageAttachmentDto } from "./MessageAttachmentDto";

// Transport shape: raw JSON from backend (dates as ISO strings)
export interface ReadMessageDto {
  id: string;
  text: string;
  userId: string;
  chatId: string;
  status: MessageStatus | number; // allow numeric flag if backend sends int
  createdAt: string; // ISO
  attachments?: MessageAttachmentDto[]; // Optional attachments array
}
